### PR TITLE
Memory Card Game

### PR DESCRIPTION
**Pull request by codal-umistri**
# Memory  card

### Components :
- Header: Added a responsive header component to display highscore and score.
- CustomInput: Created an input component for adjusting the number of cards with validation.
- DisplayModal: Implemented a modal component to show game outcomes with retry functionality.
- Card: Developed a card component with hover and flip effects for displaying Pokémon data.

### Attachment:-
image
image

### CODE REVIEW

File: src/components/Card.tsx [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  handleClick: (name: string) => void;
+  flipStatus: boolean;
+};
+
+const Card: React.FC<CardProps> = ({
+  id,
+  data,
+  handleClick,
+  flipStatus,
+}: any) => {
```
Comment (dhruv-codal) [inline src/components/Card.tsx:17]: > **fix**

To add proper types for the same

File: src/components/Modal.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  modalDisplay: boolean;
+  setModalDisplay: React.Dispatch<React.SetStateAction<boolean>>;
+}
+
+const DisplayModal: React.FC<ModalProps> = ({
+  outcome,
+  modalDisplay,
+  setModalDisplay,
+}) => {
+  if (!modalDisplay) return null;
```
Comment (dhruv-codal) [inline src/components/Modal.tsx:15]: > **note**

For `null` returns, prefer to use `consle.log` to highlight the element render

File: src/components/Cardlist.tsx [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  let randomArray = shuffleArray([...Array(cardAmount).keys()]);
+
+  if (loading) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <>
+      {pokemonData.length > 0 && (
+        <div className="cards-container">
```
Comment (dhruv-codal) [inline src/components/Cardlist.tsx:98]: > **fix**

Use grid for this

Comment (codal-umistri) [reply src/components/Cardlist.tsx:89]: Already used !

**Review by dhruv-codal — APPROVED** [review]

**Pull request merged**